### PR TITLE
[doc] [client] [go] Add chunking to go-client doc

### PR DESCRIPTION
**Pull request by Gleiphir2769**
Master Issue: pulsar-client-go [#865](https://github.com/apache/pulsar-client-go/pull/805)

### Motivation

Add chunking document for go-client. (Do not merge. [#865](https://github.com/apache/pulsar-client-go/pull/805) is under review now.)

### Modifications

Add chunking document for go-client.

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [ ] `doc-not-needed` 
(Please explain why)

- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### Matching PR in forked repository

PR in forked repository:  https://github.com/Gleiphir2769/pulsar-client-go

### CODE REVIEW

**Comment by Anonymitaet** [comment]
Thanks for your contribution! Feel free to add your updates to https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914 once the PRs get merged.

**Comment by Anonymitaet** [comment]
@RobertIndie @nodece could you please review this PR after https://github.com/apache/pulsar-client-go/pull/805 is merged? Thanks!

**Comment by Gleiphir2769** [comment]
> Thanks for your contribution! Feel free to add your updates to https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914 once the PRs get merged.

@Anonymitaet Hi, the ralated pr has merged https://github.com/apache/pulsar-client-go/pull/805. But I have no permits to edit this doc.

**Comment by Anonymitaet** [comment]
> > Thanks for your contribution! Feel free to add your updates to https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914 once the PRs get merged.
> 
> @Anonymitaet Hi, the ralated pr has merged [apache/pulsar-client-go#805](https://github.com/apache/pulsar-client-go/pull/805). But I have no permits to edit this doc.

@Gleiphir2769 can you leave comments in https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914? Then I'll copy your changes to that file, thanks!

**Comment by Gleiphir2769** [comment]
> > > Thanks for your contribution! Feel free to add your updates to https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914 once the PRs get merged.
> > 
> > 
> > @Anonymitaet Hi, the ralated pr has merged [apache/pulsar-client-go#805](https://github.com/apache/pulsar-client-go/pull/805). But I have no permits to edit this doc.
> 
> @Gleiphir2769 can you leave comments in https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914? Then I'll copy your changes to that file, thanks!

@Anonymitaet  Done, thx.

**Comment by Gleiphir2769** [comment]
@RobertIndie @nodece Hi, could you review this PR and merge it soon?

File: site2/docs/client-libraries-go.md [hunk 1]
```diff
@@ -204,6 +204,34 @@ for i := 0; i < 10; i++ {
 }
 ```
 
+#### How to use chunking in producer
+
+```go
+client, err := NewClient(pulsar.ClientOptions{
```
Comment (nodece) [inline site2/docs/client-libraries-go.md:210]:
```suggestion
client, err := pulsar.NewClient(pulsar.ClientOptions{
```

**Review by RobertIndie — APPROVED** [review]

**Review by nodece — APPROVED** [review]

File: site2/docs/client-libraries-go.md [hunk 2]
```diff
@@ -204,6 +204,34 @@ for i := 0; i < 10; i++ {
...
+	URL: serviceURL,
+})
+
+if err != nil {
+	log.Fatal(err)
+}
+defer client.Close()
+
+// The message chunking feature is OFF by default.
+// By default, producer chunks the large message based on max message size (maxMessageSize) configured at broker (eg: 5MB).
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:220]:
```suggestion
// By default, a producer chunks the large message based on the max message size (`maxMessageSize`) configured at the broker side (for example, 5MB).
```

File: site2/docs/client-libraries-go.md [hunk 3]
```diff
@@ -204,6 +204,34 @@ for i := 0; i < 10; i++ {
...
+// Client can also configure max chunked size using producer configuration ChunkMaxMessageSize.
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:221]:
```suggestion
// Client can also configure the max chunked size using the producer configuration `ChunkMaxMessageSize`.
```

File: site2/docs/client-libraries-go.md [hunk 4]
```diff
@@ -204,6 +204,34 @@ for i := 0; i < 10; i++ {
...
+// Attention, to enable chunking, you need to disable batching (DisableBatching=true) concurrently.
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-go.md:222]:
```suggestion
// Note: to enable chunking, you need to disable batching (`DisableBatching=true`) concurrently.
```

**Review by Anonymitaet — APPROVED** [review]

File: site2/docs/client-libraries-go.md (hunk 2 above)
Comment (Gleiphir2769) [reply site2/docs/client-libraries-go.md:220]: Good suggestion, thx!

File: site2/docs/client-libraries-go.md (hunk 3 above)
Comment (Gleiphir2769) [reply site2/docs/client-libraries-go.md:221]: Good suggestion, thx!

File: site2/docs/client-libraries-go.md (hunk 4 above)
Comment (Gleiphir2769) [reply site2/docs/client-libraries-go.md:222]: Good suggestion, thx!

**Pull request merged**